daemon/graphdriver/quota/projectquota.go

```diff
@@ -335,3 +352,26 @@ func makeBackingFsDev(home string) (string, error) {
 
 	return backingFsBlockDev, nil
 }
+
+func hasQuotaSupport(backingFsBlockDev string) (bool, error) {
+	var cs = C.CString(backingFsBlockDev)
+	defer free(cs)
+	var dqinfo C.struct_dqinfo
+
+	_, _, errno := unix.Syscall6(unix.SYS_QUOTACTL, uintptr(C.Q_GETINFO_PRJQUOTA), uintptr(unsafe.Pointer(cs)), 0, uintptr(unsafe.Pointer(&dqinfo)), 0, 0)
+	if errno == 0 {
+		return true, nil
+	}
+
+	switch errno {
```

Comment:
is there a specific errno you are looking for here that denotes no quota support?

Reply:
if not and you want to keep this switch i would change it to:

```go
switch errno {
    case unix.EINVAL, unix.EFAULT, etc....:
    default:
 ....
}
```

Reply:
No, there are a few errors that can denote no quota support (ENOTTY, ENODEV, ENOSUPP, etc.. -- unfortunately, they all mean different things)

Reply:
Basically, these are the fatal, non-quota related errors that can occur.

Reply:
ok, i'd probably just update the switch and put those all on one line and add a comment that you are filtering out unknown errors because there is not a single "no quota supported errno"